File: pkg/codegen/testing/test/testdata/simplified-invokes-pp/dotnet/simplified-invokes.cs
using System.Collections.Generic;
using Pulumi;
using Std = Pulumi.Std;

return await Deployment.RunAsync(() => 
{
    var everyArg = Std.AbsMultiArgs.Invoke(10, 20, 30);

    var onlyRequiredArgs = Std.AbsMultiArgs.Invoke(10);

    var optionalArgs = Std.AbsMultiArgs.Invoke(10, null, 30);

    var nestedUse = Output.Tuple(everyArg, Std.AbsMultiArgs.Invoke(42)).Apply(values =>
    {
        var everyArg = values.Item1;
        var invoke = values.Item2;
        return Std.AbsMultiArgs.Invoke(everyArg, invoke);
    });

    return new Dictionary<string, object?>
    {
        ["result"] = nestedUse,
    };
});

File: pkg/codegen/testing/test/testdata/simplified-invokes-pp/nodejs/simplified-invokes.ts
import * as pulumi from "@pulumi/pulumi";
import * as std from "@pulumi/std";

const everyArg = std.AbsMultiArgs(10, 20, 30);
const onlyRequiredArgs = std.AbsMultiArgs(10);
const optionalArgs = std.AbsMultiArgs(10, undefined, 30);
const nestedUse = Promise.all([everyArg, std.AbsMultiArgs(42)]).then(([everyArg, invoke]) => std.AbsMultiArgs(everyArg, invoke));
export const result = nestedUse;
